### PR TITLE
BaSP-MR-55: Create Subscription tests for post/get 

### PR DESCRIPTION
**Pull request by enriqueanastasini**
Create of test for subscription. I also had to change the controller of subscription because it was never returning 404 when get by id.

### CODE REVIEW

File: src/tests/subscription.test.js [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+});
+
+describe('POST /subscription', () => {
+  test('should create a new subscription', async () => {
+    const response = await request(app).post('/api/subscription').send(mockSubscription);
+
+    expect(response.status).toBe(201);
+    expect(response.error).toBeFalsy();
+  });
+});
```
Comment (EmaBaldor) [inline src/tests/subscription.test.js:24]: Missing post test with code status 400 and 500

Comment (enriqueanastasini) [reply src/tests/subscription.test.js:24]: Thank you for, i will add it

**Review by EmaBaldor — APPROVED** [review]

File: src/tests/subscription.test.js [hunk 2]
```diff
@@ -1,5 +1,89 @@
...
+    const response = await request(app).post('/api/subscription').send(mockSubscription);
+
+    expect(response.status).toBe(201);
+    expect(response.error).toBeFalsy();
+  });
+
+  test('should return 400 if required fields are missing', async () => {
+    const invalidSubscription = {
+      classId: '64667748fc13ae7f027543cc',
+      // Missing 'members' field
```
Comment (JuanMornik) [inline src/tests/subscription.test.js:28]: Delete this comment please

File: src/tests/subscription.test.js [hunk 3]
```diff
@@ -1,5 +1,89 @@
...
+      members: '6465537202739f6df0878cd1',
+      date: new Date('2022-10-29'),
+    };
+    const response = await request(app).post('/api/subscription').send(mockCreateSubscription);
+    expect(response.status).toBe(500);
+    expect(response.error).toBeTruthy();
+  });
+});
+
+describe('GET /api/subscription', () => {
```
Comment (JuanMornik) [inline src/tests/subscription.test.js:52]: Status 500 test missing

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/tests/subscription.test.js [hunk 4]
```diff
@@ -58,6 +99,32 @@ describe('PUT /api/subscription/:id', () => {
   });
 });
 
+describe('GET /subscription/:id', () => {
```
Comment (JuanMornik) [inline src/tests/subscription.test.js:102]: Missing status 500 test

**Review by JuanMornik — CHANGES_REQUESTED** [review]

**Pull request merged**